Sources/Core/AWSSDKIdentity/Sources/AWSSDKIdentity/AWSCredentialIdentityResolvers/STSAssumeRoleAWSCredentialIdentityResolver.swift

```diff
@@ -64,7 +64,10 @@ public struct STSAssumeRoleAWSCredentialIdentityResolver: AWSCredentialIdentityR
             creds: underlyingCreds,
             roleARN: roleARN,
             roleSessionName: roleSessionName,
-            durationSeconds: durationSeconds
+            durationSeconds: durationSeconds,
+            credentialFeatureIDs: underlyingCreds.properties.get(
+                key: AWSIdentityPropertyKeys.credentialFeatureIDs
+            ) ?? []
```

Comment:
The reason `credentialFeatureIDs` of credentials being used with STS assume role call is being passed to `InternalAWSSTS::assumeRole` isn't to inject it into operation context to add it into user agent header. The reason it's passed is to allow `IdentityProvidingSTSClient` to include feature IDs resolved so far in the new credentials resolved from a successful STS assume role call.